libs-scala/executors/src/main/scala/com/daml/executors/NamedExecutor.scala

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.executors.executors
+
+trait NamedExecutor {
+
+  /** Unique name for the executor service that can be used as identifier in logs and metrics
+    */
+  def name: String
```

Comment:
How about changing the type of this to `MetricName` ?

Reply:
The intention is that the self service error codes associated with rate limiting have a reference to the _metric name_.  A metric label is next to useless unless the name that is known. Additionally the same label could be applied to many metrics.

Reply:
Maybe both are needed ?

Reply:
The idea is that from the self service error code an operator can go into Grafana and display the metric with that name and see where the resource being limited is breaching the limit over time.

Reply:
Yes, so the name is already referenced, but it's not this one, it's a constant one for metrics.
Then for the name the label has to be used to know for which pool is the metrics. Only the label is dynamic and per pool.